controller/pokemon.go

```diff
@@ -0,0 +1,60 @@
+package controller
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/rmonroy-wiz/ondemand-go-bootcamp-2022/business"
+)
+
+type pokemon struct {
+	pokemonBusiness business.PokemonBusiness
+}
+
+type Pokemon interface {
+	GetAllPokemons()
+}
+
+func NewPokemonController(pokemonBusiness business.PokemonBusiness) *pokemon {
+	return &pokemon{
+		pokemonBusiness: pokemonBusiness,
+	}
+}
+func (p pokemon) GetAllPokemons(c *gin.Context) {
+	pokemons, err := p.pokemonBusiness.GetAll()
+
+	if err != nil {
+		c.AbortWithStatus(http.StatusNotFound)
```

Comment:
this is applicable for all endpoints, you must add a `return` statement after writing the response/response headers 
Gin prevents overriding the status code, but other frameworks might override the 404 with the default status code, http 200 in this case